src/layers/others.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    LayerScale(scale) 
+
+Implements LayerScale.
+([reference](https://arxiv.org/abs/2103.17239))
+
+# Arguments
+- `scale`: Scaling factor, a learnable diagonal matrix which is multiplied to the input.
+"""
+struct LayerScale
+  scale
+end
+
+"""
+    LayerScale(λ, planes::Int)
+
+Implements LayerScale.
+([reference](https://arxiv.org/abs/2103.17239))
+
+# Arguments
+- `planes`: Size of channel dimension in the input.
+- `λ`: initialisation value for the learnable diagonal matrix.
+"""
+LayerScale(planes::Int, λ) = λ > 0 ? LayerScale(Flux.ones32(planes) * λ) : identity
+
+@functor LayerScale
+(m::LayerScale)(x::AbstractArray{T}) where T = convert.(T, m.scale) .* x
```

Comment:
You don't need to manually convert here. You could enforce `Float32` in the constructor. Julia's promotion rules should be used on the forward pass.

Reply:
Ah okay I'll make that change 👍🏽